src/extension-runners/index.js

```diff
@@ -0,0 +1,67 @@
+/* @flow */
+
+import type {
+  IExtensionRunner,  // eslint-disable-line import/named
+} from './base';
+
+export type MultipleTargetsExtensionRunnerParams = {
+  runners: Array<IExtensionRunner>,
+};
+
+// Export everything exported by the firefox-desktop runner.
+export * from './firefox-desktop';
+
+export class MultipleTargetsExtensionRunner {
+  extensionRunners: Array<IExtensionRunner>;
+
+  constructor(params: MultipleTargetsExtensionRunnerParams) {
+    this.extensionRunners = params.runners;
+  }
+
+  async run(): Promise<void> {
+    const promises = [];
+    for (const runner of this.extensionRunners) {
+      promises.push(runner.run());
+    }
+
+    await Promise.all(promises);
+  }
+
+  async reloadAllExtensions(): Promise<void> {
+    const promises = [];
+    for (const runner of this.extensionRunners) {
+      promises.push(runner.reloadAllExtensions());
+    }
+
+    await Promise.all(promises);
+  }
+
+  async reloadExtensionBySourceDir(sourceDir: string): Promise<void> {
+    const promises = [];
+    for (const runner of this.extensionRunners) {
+      promises.push(runner.reloadExtensionBySourceDir(sourceDir));
+    }
+
+    await Promise.all(promises);
+  }
+
+  registerCleanup(fn: Function): void {
+    const promises = [];
+    for (const runner of this.extensionRunners) {
+      promises.push(new Promise((resolve) => {
+        runner.registerCleanup(resolve);
```

Comment:
This is basically creating a promise for every one of the created runners, each promise is going to be resolved when the related runner calls its registered cleanup functions.

The goal is be able to wait that all the runners have called their registered cleanup callbacks, so that we can finally call the cleanup callbacks registered on the "multi targets extension runner" runner.

I'll add a comment in the next update.